app/javascript/dashboard/routes/dashboard/commands/commandbar.vue

```diff
@@ -1,118 +1,108 @@
-      // Else we do nothing and its handled in the ChatList.vue hideCustomSnoozeModal() method
-      if (
-        this.selectedSnoozeType !==
-        wootConstants.SNOOZE_OPTIONS.UNTIL_CUSTOM_TIME
-      ) {
-        this.$store.dispatch('setContextMenuChatId', null);
-      }
-    },
-  },
+import { GENERAL_EVENTS } from 'dashboard/helper/AnalyticsHelper/events';
+
+const store = useStore();
+const route = useRoute();
+const track = useTrack();
+const { t } = useI18n();
+
+const ninjakeys = ref(null);
+
+// Added selectedSnoozeType to track the selected snooze type
+// So if the selected snooze type is "custom snooze" then we set selectedSnoozeType with the CMD action id
+// So that we can track the selected snooze type and when we close the command bar
+const selectedSnoozeType = ref(null);
+
+const { goToAppearanceHotKeys } = useAppearanceHotKeys();
+const { inboxHotKeys } = useInboxHotKeys();
+const { goToCommandHotKeys } = useGoToCommandHotKeys();
+const { bulkActionsHotKeys, selectedConversations } = useBulkActionsHotKeys();
+const {
+  conversationHotKeys,
+  assignableAgents,
+  currentChat,
+  teamsList,
+  activeLabels,
+  draftMessage,
+  replyMode,
+  contextMenuChatId,
+} = useConversationHotKeys();
+
+const placeholder = computed(() => t('COMMAND_BAR.SEARCH_PLACEHOLDER'));
+const routeName = computed(() => route.name);
+
+const hotKeys = computed(() => [
+  ...inboxHotKeys.value,
+  ...goToCommandHotKeys.value,
+  ...goToAppearanceHotKeys.value,
+  ...bulkActionsHotKeys.value,
+  ...conversationHotKeys.value,
+]);
+
+const setCommandBarData = () => {
+  ninjakeys.value.data = hotKeys.value;
+};
+
+const onSelected = item => {
+  const {
+    detail: { action: { title = null, section = null, id = null } = {} } = {},
+  } = item;
+  // Added this condition to prevent setting the selectedSnoozeType to null
+  // When we select the "custom snooze" (CMD bar will close and the custom snooze modal will open)
+  if (id === wootConstants.SNOOZE_OPTIONS.UNTIL_CUSTOM_TIME) {
+    selectedSnoozeType.value = wootConstants.SNOOZE_OPTIONS.UNTIL_CUSTOM_TIME;
+  } else {
+    selectedSnoozeType.value = null;
+  }
+
+  track(GENERAL_EVENTS.COMMAND_BAR, {
+    section,
+    action: title,
+  });
+
+  setCommandBarData();
+};
+
+const onClosed = () => {
+  // If the selectedSnoozeType is not "SNOOZE_OPTIONS.UNTIL_CUSTOM_TIME (custom snooze)" then we set the context menu chat id to null
+  // Else we do nothing and its handled in the ChatList.vue hideCustomSnoozeModal() method
+  if (
+    selectedSnoozeType.value !== wootConstants.SNOOZE_OPTIONS.UNTIL_CUSTOM_TIME
+  ) {
+    store.dispatch('setContextMenuChatId', null);
+  }
 };
+
+watch(
+  [
+    routeName,
+    selectedConversations,
+    assignableAgents,
+    currentChat,
+    teamsList,
+    activeLabels,
+    draftMessage,
+    replyMode,
+    contextMenuChatId,
+  ],
+  () => {
+    setCommandBarData();
+  }
+);
```

Comment:
Do we really need to watch all these?

Reply:
@scmmishra I think so, In mixins we watch all of this computed properties. In case of `routeName, selectedConversations` we must need to watch. Basically, to show data dynamically.

Reply:
Noted, let me take a look

Reply:
@iamsivin I am assuming when one of these changes the values from `useAppearanceHotKeys`, `useInboxHotKeys`, `useGoToCommandHotKeys`, `useBulkActionsHotKeys` and `useConversationHotKeys` also changes... which in turn triggers a change in the `hotKeys` computed property. 

Instead of this, can we just do this?

```js
watchEffect(() => {
  ninjakeys.value.data = hotKeys.value;
});
```

Reply:
@scmmishra Done https://github.com/chatwoot/chatwoot/pull/10015/commits/7e6fa8dd09445cb3e7b468e11b232c21bf7229d1